Lib/logging/handlers.py

```diff
@@ -638,8 +637,7 @@ def close(self):
         """
         self.acquire()
         try:
-            sock = self.sock
-            if sock:
+            if (sock := self.sock):
```

Comment:
I wonder if this code can be written as 

```
if self.sock:
    self.sock.close()
    self.sock = None
```

or is there some importance to the particular order to the setting to `None` and closing the socket? I was thinking that any race condition should be avoided due to the `self.acquire`